cf-custom-resources/lib/custom-domain-app-runner.js

```diff
@@ -0,0 +1,315 @@
+ * @returns {Promise} Promise that is resolved on success, or rejected on connection error or HTTP error response
+ */
+function report (
+    event,
+    context,
+    responseStatus,
+    physicalResourceId,
+    responseData,
+    reason
+) {
+    return new Promise((resolve, reject) => {
+        const https = require("https");
+        const { URL } = require("url");
+
+        let reasonWithLogInfo = `${reason} (Log: ${context.logGroupName}/${context.logStreamName})`;
+        var responseBody = JSON.stringify({
+            Status: responseStatus,
+            Reason: reasonWithLogInfo,
+            PhysicalResourceId: physicalResourceId || context.logStreamName,
+            StackId: event.StackId,
+            RequestId: event.RequestId,
+            LogicalResourceId: event.LogicalResourceId,
+            Data: responseData,
+        });
+
+        const parsedUrl = new URL(event.ResponseURL);
+        const options = {
+            hostname: parsedUrl.hostname,
+            port: 443,
+            path: parsedUrl.pathname + parsedUrl.search,
+            method: "PUT",
+            headers: {
+                "Content-Type": "",
+                "Content-Length": responseBody.length,
+            },
+        };
+
+        https
+            .request(options)
+            .on("error", reject)
+            .on("response", (res) => {
+                res.resume();
+                if (res.statusCode >= 400) {
+                    reject(new Error(`Error ${res.statusCode}: ${res.statusMessage}`));
+                } else {
+                    resolve();
+                }
+            })
+            .end(responseBody, "utf8");
+    });
+}
+
+exports.handler = async function (event, context) {
+    const props = event.ResourceProperties;
+    const [serviceARN, appDNSRole, customDomain] = [props.ServiceARN, props.AppDNSRole, props.CustomDomain,];
+    appHostedZoneID = props.HostedZoneID;
+    const physicalResourceID = `/associate-domain-app-runner/${customDomain}`;
+    let handler = async function () {
+        // Configure clients.
+        appRoute53Client = new AWS.Route53({
+            credentials: new AWS.ChainableTemporaryCredentials({
+                params: { RoleArn: appDNSRole, },
+                masterCredentials: new AWS.EnvironmentCredentials("AWS"),
+            }),
+        });
+        appRunnerClient = new AWS.AppRunner();
+
+        switch (event.RequestType) {
+            case "Create":
+                await addCustomDomain(serviceARN, customDomain);
+                await waitForCustomDomainToBeActive(serviceARN, customDomain);
+                break;
+            case "Update":
+            case "Delete":
+                throw new Error("not yet implemented");
+            default:
+                throw new Error(`Unsupported request type ${event.RequestType}`);
+        }
+    };
+
+    try {
+        await Promise.race([exports.deadlineExpired().catch(err => {throw err;}), handler(),]);
```

Comment:
nit: I believe this is equivalent to 
```suggestion
        await Promise.race([exports.deadlineExpired(), handler()]);
```

Reply:
ooo I thought the `throw` needs to be done explicitly! done.